pageserver/src/tenant/delete.rs

```diff
@@ -546,8 +544,7 @@ impl DeleteTenantFlow {
             conf,
             remote_storage.as_ref(),
             &tenant.tenant_shard_id,
-            // Can't use tenant.cancel, it's already shut down.  TODO: wire in an appropriate token
-            &CancellationToken::new(),
+            &task_mgr::shutdown_token(),
```

Comment:
This is a step forward, but, why not go through the trouble of plumbing through the `CancellationToken` from the two callers in this PR?

Reply:
I.e., 
<img width="455" alt="image" src="https://github.com/neondatabase/neon/assets/956573/9f1053bc-6ae6-4796-82f9-b1b9fe758d40">

Reply:
It's a bit awkward:
- If I provide a `CancellationToken` argument, callers have to be careful not to pass in the tenant's cancellation token (rather counter-intuitively) -- they need the TenantManager-scoped one, or a task scope one.
- Getting at such a token from the `spawn`->`resume_from_attach` path means passing it into `spawn`, which is also kind of confusing to read -- the argument has this special property of "please only use this cancellation token for deletions" (e.g. it would be wrong to use it as the parent for the cancellation token for the tenant being spawned).

So TLDR it's doable, but would probably come with a lot of comment-lines to explain it all at each step in the call stack.

The end state for this stuff should be simplified deletion, where we get rid of this horrible complexity that comes from sometimes spawning a deletion from the tenant load process:
- #5080